Fix PVC.Status.Capacity and AccessModes after binding

jsafrane opened the pull request:
Also, fix unit tests to have the same claim and volume sizes in most of the
tests where we don't test matching based on size and test for a specific size
when we do actually test the matching.

Fixes #29425

saad-ali commented on pkg/controller/volume/persistentvolume/controller.go:
```diff
@@ -517,14 +518,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
-		glog.V(4).Infof("updating PersistentVolumeClaim[%s]: phase %s already set", claimToClaimKey(claim), phase)
-		return claim, nil
-	}
+// updateClaimStatusWithEvent saves new claim.Status to API server.
```
`updateClaimStatus` not `updateClaimStatusWithEvent`

saad-ali replied:
Please add comment explaining what the parameters are and how they are used.

jsafrane replied:
comments added

saad-ali commented on pkg/controller/volume/persistentvolume/controller.go:
```diff
@@ -517,14 +518,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *api.PersistentVolume)
-		return claim, nil
-	}
+// updateClaimStatusWithEvent saves new claim.Status to API server.
+func (ctrl *PersistentVolumeController) updateClaimStatus(claim *api.PersistentVolumeClaim, phase api.PersistentVolumeClaimPhase, volume *api.PersistentVolume) (*api.PersistentVolumeClaim, error) {
```
As far as I can tell `updateClaimStatus(...)` is called with `volume *api.PersistentVolume` set to `nil` everywhere except by `updateClaimStatusWithEvent(...). And `updateClaimStatusWithEvent(...) is called with `volume *api.PersistentVolume` nil everywhere. So is volume ever not nil?

jsafrane replied:
`bind()` calls `updateClaimStatus` with non-nil volume.

I agree that `updateClaimStatusWithEvent` is always called with `nil`, however I'd prefer to keep the volume explicitly there for sake of consistency.

saad-ali replied:
Ack

saad-ali commented:
LGTM

saad-ali commented:
Please squash commits and add LGTM label

jsafrane committed:
Fix PVC.Status.Capacity and AccessModes after binding

Also, fix unit tests to have the same claim and volume sizes in most of the
tests where we don't test matching based on size and test for a specific size
when we do actually test the matching.

jsafrane commented:
squashed and pushed

saad-ali commented:
@jsafrane Please cherry-pick this to the 1.3 release branch. 

CC @fabioy
